lib/spack/spack/environment/environment.py

```diff
@@ -63,7 +63,10 @@
 
 
 #: path where environments are stored in the spack tree
-env_path = os.path.join(spack.paths.var_path, "environments")
+# circular dependency to use canonicalize_path but we want to ignore an active env anyways
+# env_path = spack.config.get(os.path.expandvars(os.path.expanduser("config:environments_root")))
+def env_path():
+    return spack.util.path.canonicalize_path(spack.config.get("config:environments_root"), False)
```

Comment:
The code default should be the same as the config default (i.e. `os.path.join(spack.paths.avar_path, "environments")`). We try to make the default config be only advisory anywhere we can.

I also think this line will be easier to read if we make the second argument explicit,

```suggestion
    return spack.util.path.canonicalize_path(
        spack.config.get("config:environments_root", os.path.join(spack.paths.var_path, "environments")),
        allow_env=False
    )
```